pkg/kubelet/cm/container_manager_windows.go

```diff
@@ -115,7 +124,37 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		cadvisorInterface: cadvisorInterface,
 	}
 
-	cm.topologyManager = topologymanager.NewFakeManager()
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.WindowsCPUAndMemoryAffinity) {
+		klog.InfoS("Creating topology manager")
+		cm.topologyManager, err = topologymanager.NewManager(machineInfo.Topology,
```

Comment:
Topology manager is used below with the device manager.
I think we need move the creation of that outside of the feature gate check

Reply:
but prior to this change we just use a fake topology manager...?
Does the devicemanager not need an implemented topology manager?

Reply:
(either way i think the error check on lines 155-157 in the current iteration needs to move to before we try and create the cpu manager)

Reply:
> (either way i think the error check on lines 155-157 in the current iteration needs to move to before we try and create the cpu manager)

fixed

Reply:
> but prior to this change we just use a fake topology manager...?
> Does the devicemanager not need an implemented topology manager?

yes, I miss read this when I was initially looking at it.  the fake topology manager is passed in but does nothing since it doesn't know about numa nodes:

https://github.com/kubernetes/kubernetes/blob/ead64fb8f0d343bd7ce9bc5b996c4b74d42a69d5/pkg/kubelet/cm/devicemanager/manager.go#L735-L738

This currently doesn't change that behavior since we are not turning on Memory manager in this PR.  That will come as a follow up.  It would also require the device to set a Topology setting on it https://github.com/kubernetes/kubernetes/blob/1baf7d4586671d155ed2fb579080be0726e2ca04/pkg/kubelet/cm/devicemanager/topology_hints.go#L141-L149

Reply:
ahh, thanks for the info.